Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -42,6 +44,8 @@ @interface CDTURLSessionTask ()
 
 @property (nonatomic) int remainingRetries;
 
+@property BOOL finished;
```

Comment:
Does this need to be some sort of atomic primitive? I'm not sure but it looks like it is being read from and written to by different threads.